Comprehensive UI redesign to match OpenTofu.org aesthetic

Yantrio opened the pull request:
Major visual overhaul of the registry UI to create a cohesive brand experience. Improving the layout to match what we did on opentofu.org.

Yantrio committed:
feat: comprehensive UI redesign to match OpenTofu.org aesthetic

Major visual overhaul of the registry UI to create a cohesive brand experience:

Design System Updates:
- Add fixed gradient background pattern matching OpenTofu.org
- Implement modern color scheme with improved dark mode support
- Add custom scrollbar styling for consistency

Navigation & Layout:
- Redesign navbar with transparent background and absolute positioning
- Modernize announcement bar with subtle transparent styling
- Add IDE-style terminal title bar with integrated breadcrumbs
- Update tree view navigation with cleaner indentation (no connecting lines)
- Implement tab-style language picker with underline indicators

Component Modernization:
- Enhance provider header with GitHub org avatars and metadata (stars, forks, repository)
- Consolidate version displays into single modern component
- Redesign sidebar blocks with better typography and spacing
- Update code blocks with distinct background colors for visibility
- Modernize all hover states and transitions

Information Architecture:
- Move repository link from sidebar to header for prominence
- Add clear "Latest" and "Older versions" sections in version sidebar
- Improve metadata display with consistent styling
- Better visual hierarchy throughout with refined spacing

This brings the registry UI in line with the main OpenTofu.org site design,
creating a more professional and cohesive user experience.

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
feat: modernize module UI components to match provider design

- Update module header with GitHub avatars and enriched metadata
- Add stars, forks, and repository information to header
- Modernize version info display with consistent styling
- Remove repository from sidebar (moved to header)
- Update sidebar menu with proper padding and modern styling
- Enhance tab links with rounded corners and smooth transitions
- Simplify "How to use" instructions with cleaner code styling
- Add target system display in module header

This brings the module pages in line with the provider design,
creating a consistent experience across all registry pages.

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
feat: add anchors to list items starting with code blocks in schema docs

- Created rehypeCodeListAnchors plugin to detect list items that start with inline code
- Wraps code elements in clickable anchor tags linking to themselves
- Adds IDs to list items for direct linking (e.g., #attr-stack_id)
- Handles hash scrolling on page load for direct links to schema attributes
- Useful for schema documentation where attributes are listed with code formatting

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
fix: add missing unist-util-visit and @types/hast dependencies

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
feat: add yellow highlight flash for anchor navigation

- Flash anchored elements with yellow background for 2.5 seconds
- Helps users identify the target element after navigation
- Smooth fade-in/fade-out animation for better UX

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
feat: center anchored elements in viewport when scrolling

- Changed scroll behavior from 'start' to 'center' for better visibility
- Anchored elements now appear in the middle of the screen
- Improves user experience when navigating to specific schema items

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
fix: ensure consistent provider card heights and fix virtualization gaps

- Add line-clamp-1 to CardItemTitle to prevent multi-line titles
- Update CardItemTitle component to accept className prop
- Filter out providers without versions before virtualization
- Remove conditional rendering that caused gaps during fast scrolling
- Add safe navigation for latestVersion properties

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
feat: modernize modules page to match providers design

- Add GitHub avatars to module cards
- Display metadata badges (stars, forks, target, version)
- Implement search functionality with debouncing
- Switch from Virtuoso to VirtuosoGrid for responsive grid layout
- Ensure consistent card heights with min-height and fallback text
- Add 'Forked' indicator for forked modules
- Update card styling to be more compact and informative

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
updated searchbar

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
search experience

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
Added ToC to sidebar

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
Added breadcrumbs for docs pages

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
cleanup, fixed docs loading

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
cleaned up search results

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
fixed typings for version grouping

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
clean up provider and module cards

Signed-off-by: James Humphries <james@james-humphries.co.uk>

yottta reviewed (approved):
Looks good.

I wonder if you consider the change suggested [here](https://github.com/opentofu/registry-ui/pull/338#issuecomment-3073901015).

And in the new version, for the modules/provider search, the switch from the centered search box to the detailed view for the searched modules/providers is kinda disruptive. Maybe it's only me, but I don't find it to be a good UX. 🤷 Maybe we could have the detailed view directly shown? 🤔

yottta commented:
> > Maybe we could have the detailed view directly shown? 🤔
> 
> Would you be able to elaborate or sketch out what you had in mind? I'm not sure I understand

IMO, when I start typing, I look at the field that I started typing into.
Look at this:
https://github.com/user-attachments/assets/453f4641-3d6d-465b-82d6-9c837260f7b6

Personally I find it quite disruptive when the field that I focus on moves in a glimpse.

Now that you asked, I am thinking on either leave the field as it is and when clicked/focused, the second view is shown, before starting typing. Another approach would be to change the initial text field to a button and when pressed, show directly the second view with the top left text field focused.

apparentlymart commented on frontend/src/components/Header/index.tsx:
```diff
@@ -15,51 +15,55 @@ export function Header() {
+              aria-label="OpenTofu Registry"
+              className="flex items-center transition-colors text-gray-900 hover:text-brand-500 dark:text-gray-50 dark:hover:text-brand-500"
+            >
+              <HeaderLogo />
```
Although the markup for the `HeaderLogo` component hasn't changed here (as far as I can tell?), it seems like the CSS has changed in a way that has upset the relative positioning of the "OpenTofu" logotype and the "Registry" text that appears after the logomark:

<img width="339" height="101" alt="OpenTofu Registry header logo, but the word Registry doesn't have its baseline aligned with the baseline of the OpenTofu logotype" src="https://github.com/user-attachments/assets/765d9c40-ef4d-4246-bcfe-9f4be2ef6fb7" />

Even on the current site the baselines don't quite line up, at least on my system:

<img width="339" height="101" alt="As in the previous image, except that the misalignment isn't as severe" src="https://github.com/user-attachments/assets/4e3248c6-39ea-4569-b126-286dcd9bfdb2" />

Since we have the whole thing wrapped in a link with `aria-label="OpenTofu Registry"` anyway, it might be better to incorporate the word "Registry" into the SVG image as path(s), just like we are already doing for the logotype, so that they'll always line up without relying on any CSS position trickery. 🤔

apparentlymart reviewed (commented):
This is kinda dense and so I ran out of time partway through reviewing it. I'm sharing what I have so far, but I've not read through everything yet.

Gogotchuri commented:
Generally, the more consistent the design, the smoother the experience. So here are some inconsistencies I have noticed so far.

1) Colors on active menu items are different between docs and provider/module. And honestly, neither looks appealing to me. Let's standardize that. After I have tried a few other combinations for active items, I like 
- background color: brand-300 with 40% transparency
- text color brand-800 - it is dark enough to read well under the background color, here are two old ones and a new one.
Old:
<img width="645" height="249" alt="{0EF48446-745F-46E8-9140-60AE245E6E2F}" src="https://github.com/user-attachments/assets/c10f9e1b-1663-45d4-814f-a38bbf85327d" />
<img width="671" height="178" alt="{4FFD44E1-2571-42C6-8103-D87FCDE538DF}" src="https://github.com/user-attachments/assets/7681ec3f-c91b-40d7-89b5-e7db0d34e433" />

New:
<img width="666" height="233" alt="{2119CB6C-FCC1-4C39-83A7-85B523E66062}" src="https://github.com/user-attachments/assets/0fff1e25-5c02-4ae6-8b21-89df4ce81f65" />
<img width="664" height="205" alt="{E1877A4A-2CBD-4B9F-8236-5A94D22BC97A}" src="https://github.com/user-attachments/assets/86681b2e-7b34-43cd-8d2f-cf3376c44860" />

*I haven't experimented in dark mode, but the same should work there too*

2) The fact that the "path" section, including the home icon, jumps up after going into the docs or a specific provider/module, doesn't feel smooth. Not only that, but the font size and spacing are also changing. I would keep it in the same place and the same size when applying the gray borders and background. It will mean adding additional padding on the left on the exploration page and reducing the height of the "path" section. Below are the examples:
<img width="1915" height="900" alt="{E954B58A-130E-45D2-9A0E-6DD9F858A76C}" src="https://github.com/user-attachments/assets/c2ceef25-b997-47ba-8fe5-a070a17e40b1" />
<img width="1879" height="800" alt="{F076DD5A-7F86-43FD-A340-9BB2A153ED9D}" src="https://github.com/user-attachments/assets/737391c1-2818-43ee-8071-3d9696565347" />

3) Search bars are identical except for the one in a provider filter. Grey background in dark mode is necessary, of course. But the outline colors are different for no reason. One is brand-500 and another one is brand-500 with transparency. Let's set brand-500 to the provider filter with no transparency value.

<img width="738" height="132" alt="{4F0CCBFA-2805-4F0F-B32E-5DB78CB9EF0E}" src="https://github.com/user-attachments/assets/ff8eb64f-ab48-4c36-af23-9bf272a288aa" />
<img width="750" height="115" alt="{F4F3FA17-3852-4B6D-8E03-08F81FB6A77D}" src="https://github.com/user-attachments/assets/3df4d67e-5f76-4e0d-b9d8-8deb98f64eb2" />

Yantrio committed:
Rmoeve PageSkeleton

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
cleanup PageTitle className support, added aia-label for provider searchbar

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
Fixed bad console.log in breadcrumbs

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
rework breadcrumbs to be ol and support aria-current=page correctly

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
Clean up SearchBredcrumbs

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
added aria-labels

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
Improved styling of selected list items in docs, versions and provider/module docs

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
Fixed searchbar consistency

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
Fixed layout of breadcrumbs

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio commented:
Thanks for the feedback @Gogotchuri and @apparentlymart , I've addressed those issues.

I'll try and play with the licensing information moving to the header now to see if I can find a nice approach.

Yantrio committed:
Clean up license rendering

Signed-off-by: James Humphries <james@james-humphries.co.uk>

apparentlymart reviewed (approved):
This is quite dense and hard to review thoroughly but it seems broadly good and so I'd be in favor of merging it and then dealing with any remaining concerns as separate PRs.

In particular, since the misaligned text in the logo was already present before this change (albeit not quite as bad), and since fixing it likely means adding a bunch of extra paths to an SVG asset, I'd support dealing with that particular change as a separate PR after this is landed.

Gogotchuri reviewed (approved):
I agree with @apparentlymart. Conducting a comprehensive review of this change line by line, with 4,000 lines of front-end code changes, is challenging. However, since most of the changes concern the UI, this version feels and looks better than the current one.
I am for merging this. Thanks for doing this work @Yantrio!

Yantrio commented:
I know that @yottta expressed some concern over how jarring the new landing search experience is so if we are to merge this, I would like to do the following:

Have a child PR from this branch, in which I work on the search experience. We can review that separately and then merge all the way down into main.

(Sidenote: I just want to play with the mermaid gitgraph feature here to explain :D )

```mermaid
gitGraph
   commit id: "initial"
   
   branch cleanup-design
   checkout cleanup-design
   commit id: "design updates"
   commit id: "more cleanup"
   
   branch rework-search-landing
   checkout rework-search-landing
   commit id: "search experience improvements"
   
   checkout cleanup-design
   merge rework-search-landing
   
   checkout main
   merge cleanup-design
```

Does this approach sound okay @opentofu/maintainers ?
